_posts/2020-10-28-push-it-to-the-limit.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title: Push It To The Limit
+categories:
+- blog
+tags:
+- MUD
+- Open Source
+- Optimization
+- Firebase
+- Lich
+---
+
+I've been hosting portions of the Lich framework I help maintain on [Firebase](https://firebase.google.com/) using a free tier plan for several years now. I became inured to the monthly e-mails reminding me that I was at 90% of my data limit showing up on the ~~30th~~ ~~29th~~ ~~28th~~... shit.
+
+---
+
+## Punished By Success
+
+It was probably inevitable that we would reach this point, adoption of our project had been growing steadily and user behavior was changing in ways that increased their data demands. I wasn't going to start giving google nickles of my money each month without a fight so I prepared to spend hours of my time preventing this tragedy. To provide more context I will first answer the obvious question, why does a MUD script repository need a database?
+
+We store two pieces of data in our database, the first will probably be the source of a blog post on it's own as it is an interesting look at 'learning' from crowd sourced data however actual user demand on that data is low since users only push changes to it when their scripts notice errors in local data. The second set of data is today's troublemaker. A fair portion of the magic and transportation for the MUD in question is dependent on time, specifically in game time and the location of celestial bodies (The sun and the planet's three moons). Sadly the state of these bodies is not strictly deterministic for the players of the game until someone finds the motivation to track months worth of data and try to find the cycle that they operate on. Lacking that we can make approximations that are probably accurate within 5~ minutes based on in game observations by the one class capable of observing the celestial bodies accurately. This information is highly useful for the class in question and mildly convenient for everyone else, but takes work to obtain in game so we decided to set up an automated system for tracking these events so anyone who was interested could have it available immediately.
```

Comment:
This entire paragraph is only six sentences. In other words, some of these sentences are very long.

I'm not saying you have to change anything, just be aware. On the other hand, since I work to make my sentences as short as possible I've been told they're all the same size, so 🤷

Reply:
Visually, here's what Hemingway reports for that paragraph:

<img width="489" alt="Hemingway_Editor_bITLye4Nb6" src="https://user-images.githubusercontent.com/1181607/97442119-de493980-18ff-11eb-9132-74c54e54bf7b.png">

Reply:
Neat tool, I stream of consciousness wrote this this morning, that's when I'm at my most unintelligible.